src/states/common/approved.rs

```diff
@@ -437,6 +437,45 @@ pub trait Approved: Base {
         };
     }
 
+    fn handle_online_event(
+        &mut self,
+        payload: OnlinePayload,
+        outbox: &mut dyn EventBox,
+    ) -> Result<(), RoutingError> {
+        if !self.chain().can_add_member(payload.p2p_node.public_id()) {
+            info!("{} - ignore Online: {:?}.", self, payload);
+        } else {
+            info!("{} - handle Online: {:?}.", self, payload);
+
+            let pub_id = *payload.p2p_node.public_id();
+            self.chain_mut()
+                .add_member(payload.p2p_node.clone(), payload.age);
+            self.chain_mut().increment_age_counters(&pub_id);
+            self.handle_member_added(payload, outbox)?;
+        }
+
+        Ok(())
+    }
+
+    fn handle_offline_event(
+        &mut self,
+        pub_id: PublicId,
+        outbox: &mut dyn EventBox,
+    ) -> Result<(), RoutingError> {
+        if !self.chain().can_remove_member(&pub_id) {
```

Comment:
just to confirm, will an Adult node pass this check and carries out the block of actions below?

Reply:
yes, it only depends on the chain. note that this code existed in both adults.rs and elder/mod.rs before, so it should not be a behaviour change.